ffi/blitbuffer.lua

```diff
@@ -1826,6 +1826,79 @@ function BB_mt.__index:writePNGFromBGR(filename)
     C.free(cdata)
 end
 
+
+local function write_uint32(of, data)
+    of:write(string.char(band(data, 255)))
+    data = math.floor(data / 256)
+    of:write(string.char(band(data, 255)))
+    data = math.floor(data / 256)
+    of:write(string.char(band(data, 255)))
+    data = math.floor(data / 256)
+    of:write(string.char(data))
+end
+
+function BB_mt.__index:writeBMP(filename)
+    local output_channels = 3
+    local w, h = self:getWidth(), self:getHeight()
+    local stride = w * output_channels
+
+    local of, err = io.open(filename, "wb")
+    if err ~= nil then
+        return err
+    end
+
+    local bbdump = BB.new(w, h, TYPE_BBRGB24, nil)
+    bbdump:blitFrom(self)
+    local data = ffi.cast("unsigned char *", bbdump.data)
```

Comment:
You *should* be able to use self.data directly if self:getType() is BBRGB24 or BBRGB32

(In which case, keep track of this optional intermediate buffer, because of the final explicit free).

Reply:
(If you want to go the extra mile, you can make type-speciifc functions like writePNG, as there's possibly a Grayscale 8bpp BMP pixel-format, but it may or may not require building a palette).

Reply:
I see that this would save a `blitFrom(self)`, but would that help with the timing?

I have a `logger.dbg` after `local data = ffi.cast`. The message appears immediately, so the bbdump is not the cause.

Reply:
Probably not, because in these cases, the type matches, so the blit will be fast (sub 10ms).

But it might help alleviate pressure on the JIT nonetheless ;).